### PR TITLE
Update+ improvements  Usagesummary.ps1

### PR DESCRIPTION
**Pull request by CarlosGon**
Added 3 missing metrics.
Changed the logic for retrieving usage:
. Created an array with the list of all subscriptions
. Changed the get-azsSubscriberUsage to include the subscriberID. 
This way we will have better performance, avoiding gateway timeout errors while executing the sql query.

### CODE REVIEW

**Comment by CarlosGon** [comment]
any feedback on this?

**Comment by CarlosGon** [comment]
#sign-off

**Comment by CarlosGon** [comment]
any update on this PR?

File: Usage/Usagesummary.ps1 [hunk 1]
```diff
@@ -27,6 +27,9 @@ Param
 
 #Initialise meter hashtable
 $meters = @{
+    '578ae51d-4ef9-42f9-85ae-42b52d3d83ac' = 'ActualPremiumSnapshotSize Unit: GB*month'
+    '108fa95b-be0d-4cd9-96e8-5b0d59505df1' = 'ActualStandardSnapshotSize Unit: GB*month'
+    'Custom Worker Tiers' = 'Custom Worker Tiers Unit: Hours'
```
Comment (cblackuk) [inline Usage/Usagesummary.ps1:32]: Can you format this please? :-)

File: Usage/Usagesummary.ps1 [hunk 2]
```diff
@@ -111,46 +114,51 @@ $meters = @{
 }
 
 #Build a subscription hashtable
-$subtable = @{}
+$subtable = New-Object System.Collections.ArrayList
```
Comment (cblackuk) [inline Usage/Usagesummary.ps1:117]: Why?

File: Usage/Usagesummary.ps1 [hunk 3]
```diff
@@ -111,46 +114,51 @@ $meters = @{
...
-    $usageSummary += $record
+    foreach ($entry in $result) {
+        $resourceInfo = ($entry.InstanceData | ConvertFrom-Json).'Microsoft.Resources'
+        $resourceText = $resourceInfo.resourceUri
+        $subscription = $resourceText.Split('/')[2]
+        $resourceType = $resourceText.Split('/')[7]
+        $resourceGroup = $resourceText.Split('/')[4]
+        $resourceName = $resourceText.Split('/')[8]
+
+        [void]($usageSummary.Add([pscustomobject] @{
```
Comment (cblackuk) [inline Usage/Usagesummary.ps1:143]: Can you format doc in VSCode to make it a little more readable?

File: Usage/Usagesummary.ps1 [hunk 4]
```diff
@@ -111,46 +114,51 @@ $meters = @{
...
-    $record | Add-Member -Name resourceName -MemberType NoteProperty -Value $resourceName
-    $record | Add-Member -Name subowner -MemberType NoteProperty -Value $subtable[$subscription]
-    $record | Add-Member -Name tags -MemberType NoteProperty -Value $resourceInfo.tags
-    $record | Add-Member -Name MeterId -MemberType NoteProperty -Value $_.MeterId
-    $record | Add-Member -Name additionalInfo -MemberType NoteProperty -Value $resourceInfo.additionalInfo
-    $record | Add-Member -Name subscription -MemberType NoteProperty -Value $subscription
-    $record | Add-Member -Name resourceUri -MemberType NoteProperty -Value $resourceText
-    $usageSummary += $record
+    foreach ($entry in $result) {
+        $resourceInfo = ($entry.InstanceData | ConvertFrom-Json).'Microsoft.Resources'
```
Comment (cblackuk) [inline Usage/Usagesummary.ps1:136]: Can you format doc in VSCode to make it a little more readable?

**Review by cblackuk — CHANGES_REQUESTED** [review]
Good changes, but need a few tiny fixes - IMHO :-)

**Review by pappleby64 — APPROVED** [review]
Look like good changes - agree with other comments

**Comment by bganapa** [comment]
@CarlosGon could you pls address the comments? Can you pls make the PR to the az branch as well?

**Comment by CarlosGon** [comment]
@bganapa 
Comments resolved. Code formatted with VScode

**Pull request merged**